Social: Add support section and text with icon component

gmjuhasz opened the pull request:
This change adds a new Text with Icon component, and places one into the support section. This should only be visible to paying users.

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Added component
* Created new support section
* Conditionally show the new component for paying users

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

1202726932197503-as-1202670259981508

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

No.

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

Testing component:
* Component should match [design](jpldgotrSl8JMHjgUXZLZY-fi-5802%3A39131)
* Check full size and small size screen version too

Testing conditional hiding:
* Fire up jetpack and start Jurassic Tube tunneling. Opening up the site, the support component should not be visible.
* Head to https://wordpress.com/checkout/jetpack_social_monthly and select your Jurassic Tube site
* Finishing up the payment, the section should appear on the admin page.
* Deleting the subscription from My Jetpack should hide it again.

<img width="1220" alt="CleanShot 2022-08-15 at 11 11 38@2x" src="https://user-images.githubusercontent.com/36671565/184608669-3a3e97b7-6753-4327-b4c5-4719f436c034.png">

gmjuhasz committed:
[not verified] Add component to admin page

gmjuhasz committed:
Only show to paid users

gmjuhasz committed:
changelog

gmjuhasz committed:
Move condition

danielpost commented on projects/plugins/social/src/js/components/icon-text-section/index.js:
```diff
@@ -0,0 +1,16 @@
+import styles from './styles.module.scss';
+
+const IconTextSection = ( { icon, title, children } ) => (
+	<Container horizontalSpacing={ 7 } horizontalGap={ 3 }>
```
What do you think about moving the `Container` wrapper to the `SupportSection` component and renaming this one to `IconText`? That way this component just handles the icon and text and doesn't concern itself with any layout formatting.

gmjuhasz replied:
Good idea! I did this.

danielpost commented on projects/plugins/social/src/js/components/icon-text-section/styles.module.scss:
```diff
@@ -0,0 +1,58 @@
+	column-gap: calc( var( --spacing-base ) * 3 );
+	row-gap: calc( var( --spacing-base ) * 2 );
+
+	@supports ( display: grid ) {
+		> * + * {
+			margin-top: 0;
+		}
+	}
```
For this to do anything, I think you also want to add some margin in case the browser doesn't support grid.

```css
> * + * {
	margin-top: calc( var( --spacing-base ) * 2 );
}

@supports ( display: grid ) {
	> * + * {
		margin-top: 0;
	}
}
```

You can simulate the non-grid behavior by toggling it off in the CSS Inspector in your browser.

gmjuhasz replied:
Done!

danielpost commented on projects/plugins/social/src/js/components/icon-text-section/styles.module.scss:
```diff
@@ -0,0 +1,58 @@
+		}
+	}
+
+	@media ( min-width: 600px ) {
```
I think it would be good if we use [useBreakpointMatch](https://github.com/Automattic/jetpack/tree/trunk/projects/js-packages/components/components/layout/use-breakpoint-match) here as well (and in all the other media queries).

gmjuhasz replied:
Done!

danielpost commented on projects/plugins/social/src/js/components/icon-text-section/styles.module.scss:
```diff
@@ -0,0 +1,58 @@
+
+	justify-self: flex-end;
+
+	margin-right: calc( var( --spacing-base ));
```
```suggestion
	margin-right: var( --spacing-base );
```

I believe `calc` is redundant here.

gmjuhasz replied:
Great point ! I was playing with multiplying it by 2, but after it I forgot to remove the calc.

danielpost commented on projects/plugins/social/src/js/components/support-section/styles.module.scss:
```diff
@@ -0,0 +1,3 @@
+.link {
+	margin-top: 16px;
```
```suggestion
	margin-top: calc( var( --spacing-base ) * 2 );
```

danielpost commented on projects/plugins/social/src/js/store/selectors/jetpack-settings.js:
```diff
@@ -2,6 +2,7 @@ const jetpackSettingSelectors = {
 	getPublicizeModuleStatus: state => state.jetpackSettings,
 	isModuleEnabled: state => state.jetpackSettings.publicize_active,
 	isUpdatingJetpackSettings: state => state.jetpackSettings.is_updating,
+	isPremiumPlan: state => ! state.showNudge,
```
What do you think about renaming this to `hasPaidPlan`? Premium might indicate a specific plan in the future.

gmjuhasz replied:
Makes sense

danielpost commented on projects/plugins/social/src/js/components/support-section/index.js:
```diff
@@ -0,0 +1,33 @@
+
+	return isPremiumPlan ? (
+		<IconTextSection
+			icon={ <Icon icon={ lifesaver } size={ 30 }></Icon> }
```
I think it might be nice to use an early return here to avoid the ternary, and make it easier to add more conditions down the road.

```js
if ( ! isPremiumPlan ) {
	return null;
}

return <IconTextSection ...
```

danielpost commented on projects/plugins/social/src/js/components/support-section/index.js:
```diff
@@ -0,0 +1,33 @@
+				) }
+			</Text>
+			<Text className={ styles.link }>
+				<ExternalLink href="https://wordpress.com/support">
```
We need to add a Jetpack Redirect here (PCYsg-pY7-p2)

danielpost reviewed (commented):
Nice work! It tests well and looks great. I've added some suggestions in my review, let me know what you think.

gmjuhasz committed:
Fix review comments

gmjuhasz commented:
> Nice work! It tests well and looks great. I've added some suggestions in my review, let me know what you think.

Thanks! Good points in the comments, I addressed them!

gmjuhasz committed:
Add margin non-grid

danielpost commented on projects/plugins/social/src/js/components/icon-text/styles.module.scss:
```diff
@@ -0,0 +1,62 @@
+.column {
+	grid-column: 1 / -1;
+	display: grid;
+	grid-template-columns: auto 1fr;
+	grid-template-rows: repeat( 2, auto );
+	column-gap: calc( var( --spacing-base ) * 3 );
+	row-gap: calc( var( --spacing-base ) * 2 );
+
+	> * + * {
+		margin-top: calc( var( --spacing-base ) * 2 );
+	}
+	
+	@supports ( display: grid ) {
+		> * + * {
+			margin-top: 0;
+		}
+	}
```
I only just thought about this (and don't feel the need to change it), but this is also a good use case for CSS variables:

```scss
.column {
	--item-spacing: calc( var( --spacing-base ) * 2 );

	> * + * {
		margin-top: var( --item-spacing );
	}

	@supports ( display: grid ) {
		--item-spacing: 0;
	}
}
```
